azure-pipelines.yml

```diff
@@ -89,12 +89,12 @@ jobs:
 #
 # Debug builds
 #
-- ${{ if and(eq(variables['System.TeamProject'], 'public'), eq(variables['Build.DefinitionName'], 'coreclr-ci')) }}:
+- ${{ if eq(variables['System.TeamProject'], 'public') }}:
   - template: eng/platform-matrix.yml
     parameters:
       jobTemplate: build-job.yml
       buildConfig: debug
-      ${{ if eq(variables['Build.Reason'], 'PullRequest') }}:
+      ${{ if and(eq(variables['Build.Reason'], 'PullRequest'), eq(variables['Build.DefinitionName'], 'coreclr-ci')) }}:
         platforms:
         - Windows_NT_x64
         - Windows_NT_x86
```

Comment:
Yes, this was my intention - to simplify triggers instead of them to be "too" specific. 

We have only one build definition running in CI right now. Although we have discussed plans to have more than one CI build definition but it's not clear if we end up with two or BatchedCI will be sufficient.

Reply:
Seems like we'll end up doing many more "debug" builds than we actually need or want in "CI" jobs?

Reply:
> Seems like we'll end up doing many more "debug" builds than we actually need or want in "CI" jobs?

@BruceForstall It will be the same number of debug jobs that we have right now, right?

If we want to reduce these let's do this but as a separate change - I don't want to mix up the change to "api" with a change to the system behaviour .